### PR TITLE
Support open mail app

### PR DESCRIPTION
**Pull request by tung2744**
ref https://github.com/authgear/authgear-server/issues/2785

Screenshot_1673943550

### CODE REVIEW

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

File: sdk/src/main/java/com/oursky/authgear/EmailClient.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.oursky.authgear
+
+data class EmailClient(val packageName: String) {
+    companion object {
+        @JvmStatic val gmail = EmailClient("com.google.android.gm")
```
Comment (louischan-oursky) [inline sdk/src/main/java/com/oursky/authgear/EmailClient.kt:5]: Constants should be named in UPPER_SNAKE_CASE. See https://kotlinlang.org/docs/coding-conventions.html#property-names

File: javasample/src/main/java/com/oursky/authgeartest/MainViewModel.java [hunk 2]
```diff
@@ -261,6 +265,24 @@ public void sendWechatAuthRequest(String state) {
                 req.state = state;
                 wechatAPI.sendReq(req);
             }
+
+            @Override
+            public void onOpenEmailClient(@NonNull Context context) {
+                Intent intent = Authgear.Companion.makeEmailClientIntentChooser(
```
Comment (louischan-oursky) [inline javasample/src/main/java/com/oursky/authgeartest/MainViewModel.java:271]: Authgear.Companion does not follow Java's convention. Add `@JvmField` to makeEmailClientIntentChooser to make it directly accessible under Authgear.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Comment by tung2744** [comment]
@louischan-oursky Updated, thanks!

**Pull request merged**